src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -465,8 +567,15 @@ private function createNewInvoice(Order $order, Cart $cart, $isFreeShipping)
         $taxCalculator = $carrier->getTaxCalculator($invoice_address);
 
         // @todo: use https://github.com/PrestaShop/decimal to compute prices and taxes
-        $invoice->total_paid_tax_excl = Tools::ps_round((float) $cart->getOrderTotal(false, $totalMethod), 2);
-        $invoice->total_paid_tax_incl = Tools::ps_round((float) $cart->getOrderTotal(true, $totalMethod), 2);
+        $precision = $this->getPrecisionFromCart($cart);
+        $invoice->total_paid_tax_excl = Tools::ps_round(
+            (float) $cart->getOrderTotal(false, $totalMethod),
```

Comment:
Be careful I think here you need to use the new products as well
The new invoice can be created after one already existed, in which case you don't want the newly created invoice to contain the total of the previous invoice

Reply:
Thanks. I check it ;)